a3p-integration/proposals/z:acceptance/test-lib/sync-tools.js

```diff
@@ -0,0 +1,285 @@
+  let retries = 0;
+
+  while (retries < maxRetries) {
+    try {
+      const result = await operation();
+      log('RESULT', result);
+      if (condition(result)) {
+        return result;
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        log(`Error: ${error.message}`);
+      } else {
+        log(`Unknown error: ${String(error)}`);
+      }
+    }
+
+    retries++;
+    console.log(
+      `Retry ${retries}/${maxRetries} - Waiting for ${retryIntervalMs}ms for ${message}...`,
+    );
+    await sleep(retryIntervalMs, { log, setTimeout });
+  }
+
+  throw Error(`${message} condition failed after ${maxRetries} retries.`);
+};
+
+/**
+ * Making sure a core-eval resulted succesfully deploying a contract
+ */
+const makeGetInstances = follow => async () => {
+  const instanceEntries = await follow(
+    '-lF',
+    `:published.agoricNames.instance`,
+  );
+
+  return Object.fromEntries(instanceEntries);
+};
+
+/**
+ *
+ * @param {string} contractName
+ * @param {{follow: () => object, setTimeout: (object) => void}} ambientAuthroity
+ * @param {RetyrOptions} options
+ * @returns
+ */
+export const waitUntilContractDeplyed = (
+  contractName,
+  ambientAuthroity,
+  options,
+) => {
+  const { follow, setTimeout } = ambientAuthroity;
+  const getInstances = makeGetInstances(follow);
+  const {
+    maxRetries = 6,
+    retryIntervalMs = 3500,
+    log = console.log,
+    errorMessage = 'Error',
+  } = options;
+
+  return retryUntilCondition(
+    getInstances,
+    instanceObject => Object.keys(instanceObject).includes(contractName),
+    errorMessage,
+    // @ts-ignore
+    { maxRetries, retryIntervalMs, log, setTimeout },
+  );
+};
+
+// Making sure an account is successfully funded with vbank assets like IST, BLD etc.
+
+const makeQueryCosmosBalace = queryCb => async dest => {
+  const conins = await queryCb('bank', 'balances', dest);
+  return conins.balances;
+};
+
+/**
+ *
+ * @param {Array} balances
+ * @param {CosmosBalanceThresold} thresold
+ * @returns {boolean}
+ */
+const checkCosmosBalance = (balances, thresold) => {
+  const balance = [...balances].find(({ denom }) => denom === thresold.denom);
+  return Number(balance.amount) >= thresold.value;
+};
+
+/**
+ * @param {string} destAcct
+ * @param {{query: () => Promise<object>, setTimeout: (object) => void}} ambientAuthroity
+ * @param {{denom: string, value: number}} threshold
+ * @param {RetyrOptions} options
+ * @returns
+ */
+export const waitUntilAccountFunded = (
+  destAcct,
+  ambientAuthroity,
+  threshold,
+  options,
+) => {
+  const { query, setTimeout } = ambientAuthroity;
+  const queryCosmosBalance = makeQueryCosmosBalace(query);
+  const {
+    maxRetries = 6,
+    retryIntervalMs = 3500,
+    log = console.log,
+    errorMessage = 'Error',
+  } = options;
+
+  return retryUntilCondition(
+    async () => queryCosmosBalance(destAcct),
+    balances => checkCosmosBalance(balances, threshold),
+    errorMessage,
+    // @ts-ignore
```

Comment:
Two things: 
1. We prefer `ts-expect-error`, since it tells you when it's no longer needed, and we try to always include a reason
```suggestion
  // @ts-expect-error Casting
```

and 2.  it's unnecessary. I checked out the PR to find out why the `ts-ignore` was needed, since Error is marked optional. It turns out that in my IDE, the `ts-ignore` gets a red underline.  It's worth setting up your IDE so the TS language server reports on syntax errors. It'll also report on many typos, or at least my IDE does.

Reply:
resolved in https://github.com/Agoric/agoric-sdk/pull/10171/commits/d3f245300f2d42b83bc761c1e22950feeacd63db